pkg/cli/debug/debug.go

```diff
@@ -594,6 +594,25 @@ func (o *DebugOptions) getContainerImageCommand(pod *corev1.Pod, container *core
 func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*corev1.Pod, []string) {
 	pod := o.Attach.Pod
 
+	if len(o.ToNamespace) > 0 {
+		svcAcctOrig := pod.Spec.ServiceAccountName
+		if !o.IsNode {
```

Comment:
I don't think this is appropriate..  Instead, have ToNamespace trigger defaulting on specific skips - --keep-volumes, --keep-service-account.  Those would be defaults, if the user then specifies the flag that should override the default.

You should not have to skip volumes that are not namespace scoped, so that's a further nuance.

The warning should be when we clear things without you explicitly giving us the flag (and warnings should always have a prefix `warning: `) - i.e.:

```
warning: Volumes that referenced the source namespace have been disabled, use --keep-volumes to override this
```

and 

```
warning: Service account name cleared, pass --keep-service-account to preserve this value
```